ton_client/src/abi/decode_message.rs

```diff
@@ -58,6 +75,101 @@ impl DecodedMessageBody {
             header,
         })
     }
+
+    pub(crate) fn decode(
+        abi: &AbiContract,
+        responsible: Option<&ResponsibleCall>,
+        body: SliceData,
+        is_internal: bool,
+        internal_dst: Option<&MsgAddressInt>,
+        allow_partial: bool,
+    ) -> ClientResult<Self> {
+        match (internal_dst, responsible) {
+            (Some(internal_dst), Some(ref responsible))
+                if is_internal && internal_dst == responsible.src =>
+            {
+                let mut body = body.clone();
+
+                let receiver_func_id = body.get_next_u32().map_err(|err| {
+                    Error::invalid_message_for_decode(format!(
+                        "Can't decode function header: {}",
+                        err
+                    ))
+                })?;
+
+                if receiver_func_id == responsible.answer_id {
+                    let tokens = TokenValue::decode_params(
```

Comment:
Use `responsible.function.decode_output`. I disabled function id check for internal output messages